numaprom/udf/inference.py

```diff
@@ -35,71 +46,74 @@ def _run_model(
     payload.set_win_arr(recon_err.numpy())
     payload.set_status(Status.INFERRED)
     payload.set_metadata("version", artifact_data.extras.get("version"))
+    return payload
 
-    return "postprocess", orjson.dumps(payload, option=orjson.OPT_SERIALIZE_NUMPY)
-
-
-@conditional_forward
-def inference(_: str, datum: Datum) -> List[Tuple[str, bytes]]:
-    _start_time = time.perf_counter()
-
-    _in_msg = datum.value.decode("utf-8")
-    payload = StreamPayload(**orjson.loads(_in_msg))
-
-    _LOGGER.debug("%s - Received Payload: %r ", payload.uuid, payload)
-
-    metric_config = get_metric_config(payload.composite_keys["name"])
-    model_config = metric_config["model_config"]
-
-    train_payload = {
-        "uuid": payload.uuid,
-        **payload.composite_keys,
-        "model_config": model_config["name"],
-        "resume_training": False,
-    }
-
-    if payload.status == Status.ARTIFACT_NOT_FOUND:
-        _LOGGER.info(
-            "%s - No artifact found in prev vtx, sending to trainer. Trainer payload: %s",
-            payload.uuid,
-            train_payload,
-        )
-        return [("train", orjson.dumps(train_payload))]
 
+def _get_model(payload: StreamPayload, model_config: dict) -> Optional[ArtifactData]:
+    print(model_config)
     artifact_data = load_model(
         skeys=[payload.composite_keys["namespace"], payload.composite_keys["name"]],
         dkeys=[model_config["model_name"]],
     )
-
     if not artifact_data:
+        payload.set_status(Status.ARTIFACT_NOT_FOUND)
         _LOGGER.info(
-            "%s - No model found, sending to trainer. Trainer payload: %s",
+            "%s - Model not found for %s",
             payload.uuid,
-            train_payload,
+            payload.composite_keys,
         )
-        return [("train", orjson.dumps(train_payload))]
-
+        return None
     _LOGGER.debug("%s - Successfully loaded model from mlflow", payload.uuid)
+    return artifact_data
 
-    messages = []
 
+def _is_model_stale(
```

Comment:
this can also be a common method I think

Reply:
We only need to check for staleness in the main model.